movieHW/MainModule/Presenter/MoviePresenter.swift

```diff
@@ -0,0 +1,58 @@
+// MoviePresenter.swift
+// Copyright © RoadMap. All rights reserved.
+
+import Foundation
+
+/// Представление фильма
+final class MoviePresenter: MovieViewPresenterProtocol {
+    // MARK: - Constants
+
+    private enum Constants {
+        static let urlString =
+            "https://api.themoviedb.org/3/movie/popular?api_key=c4c9f43f1e9f5412ba0efa79f720438e&language=en-US&page=1"
+        static let resultsType = "results"
+    }
+    
+    // MARK: - Public Properties
+
+    var movies: [Movie]?
+
+    // MARK: - Private Properties
+
+    private let view: MovieViewProtocol?
+    private let networkService: NetworkServiceProtocol?
+    private var router: RouterProtocol?
+
+    // MARK: - Initializers
+
+    required init(view: MovieViewProtocol, networkService: NetworkServiceProtocol, router: RouterProtocol) {
+        self.view = view
+        self.networkService = networkService
+        self.router = router
+        getMovies()
+    }
+
+    // MARK: - Public Methods
+
+    func tapOnTheMovie(movie: Movie?) {
+        router?.showDetail(movie: movie)
+    }
+
+    func getMovies() {
+        networkService?.getMovies(
+            urlString: Constants.urlString,
+            complition: { [weak self] result in
+                guard let self = self else { return }
+                DispatchQueue.main.async {
```

Comment:
```suggestion
                DispatchQueue.main.async {
```
мы точно после AF на в главном потоке?